tests/appsec/test_automated_login_events.py

```diff
@@ -1347,3 +1223,665 @@ def validate_iden(meta):
+    """
+    Test login success/failure use cases
+    By default, mode is identification
+    """
+
+    # User entries in the internal DB:
+    # users = [
+    #     {
+    #         id: 'social-security-id',
+    #         username: 'test',
+    #         password: '1234',
+    #         email: 'testuser@ddog.com'
+    #     },
+    #     {
+    #         id: '591dc126-8431-4d0f-9509-b23318d3dce4',
+    #         username: 'testuuid',
+    #         password: '1234',
+    #         email: 'testuseruuid@ddog.com'
+    #     }
+    # ]
+    #
+    # These users can be created with signup events:
+    # users = [
+    #     {
+    #         id: 'new-user',
+    #         username: 'testnew',
+    #         password: '1234',
+    #         email: 'testnewuser@ddog.com'
+    #     }
+    # ]
+
+    def setup_login_pii_success_local(self):
+        self.r_pii_success = weblog.post("/login?auth=local", data=login_data(context, USER, PASSWORD))
+
+    def test_login_pii_success_local(self):
+        assert self.r_pii_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_pii_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.success.usr.login"] == USER
+            assert meta["_dd.appsec.usr.login"] == USER
+            assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "identification"
+            assert meta["appsec.events.users.login.success.track"] == "true"
+
+            # optional (to review for each library)
+            assert meta["usr.id"] == "social-security-id"
+            assert meta["_dd.appsec.usr.id"] == "social-security-id"
+
+    def setup_login_pii_success_basic(self):
+        self.r_pii_success = weblog.get("/login?auth=basic", headers={"Authorization": BASIC_AUTH_USER_HEADER})
+
+    @missing_feature(context.library == "php", reason="Basic auth not implemented")
+    def test_login_pii_success_basic(self):
+        assert self.r_pii_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_pii_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.success.usr.login"] == USER
+            assert meta["_dd.appsec.usr.login"] == USER
+            assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "identification"
+            assert meta["appsec.events.users.login.success.track"] == "true"
+
+            # optional (to review for each library)
+            assert meta["usr.id"] == "social-security-id"
+            assert meta["_dd.appsec.usr.id"] == "social-security-id"
+
+    def setup_login_success_local(self):
+        self.r_success = weblog.post("/login?auth=local", data=login_data(context, UUID_USER, PASSWORD))
+
+    def test_login_success_local(self):
```

Comment:
why are we doing twice the same test, one with uuid and other with username?

Reply:
I'm not sure, it was inherited from V1 and V2 and I didn't want to remove it (not sure if it's testing something special).

Reply:
I think that it was because the initial spec says:

> safe  — The safe mode must avoid collecting PIIs by collecting only non-sensitive data types and formats, i.e. internal numeric identifiers or a random sequence of characters (e.g. UUID, GUID) used to identify accounts inside the authentication framework. If no data is available, send empty events.

But now, we are not applying that, by default we are sending all the data and in safe mode, we are always anonymizing it.

Reply:
we should definitely remove the "pii" test as they're not relevant anymore

Reply:
i think this is resolved now ?

Reply:
Yep solved, thanks!